controllers/component_build_controller_pac.go

```diff
@@ -654,37 +662,57 @@ func (r *ComponentBuildReconciler) UnconfigureRepositoryForPaC(log logr.Logger,
 			}
 		}
 
-		prData := &github.PaCPullRequestData{
-			Owner:         owner,
-			Repository:    repository,
-			CommitMessage: commitMessage,
-			Branch:        branch,
-			BaseBranch:    baseBranch,
-			PRTitle:       mrTitle,
-			PRText:        mrText,
-			AuthorName:    authorName,
-			AuthorEmail:   authorEmail,
-			Files: []github.File{
-				{FullPath: ".tekton/" + component.Name + "-" + pipelineRunOnPushFilename},
-				{FullPath: ".tekton/" + component.Name + "-" + pipelineRunOnPRFilename},
-			},
+		if baseBranch == "" {
+			baseBranch, err = github.GetDefaultBranch(ghclient, owner, repository)
+			if err != nil {
+				return "", "", nil
+			}
 		}
-		prUrl, err = github.UndoPaCPullRequest(ghclient, prData)
+
+		sourceBranch := generateMergeRequestSourceBranch(component)
+		pullRequest, err := github.FindUnmergedOnboardingMergeRequest(ghclient, owner, repository, sourceBranch, baseBranch, owner)
 		if err != nil {
-			// Handle case when GitHub application is not installed for the component repository
-			if strings.Contains(err.Error(), "Resource not accessible by integration") {
-				return "", fmt.Errorf(" Pipelines as Code GitHub application with %s ID is not installed for %s repository",
-					string(config[gitops.PipelinesAsCode_githubAppIdKey]), component.Spec.Source.GitSource.URL)
+			return "", "", err
+		}
+
+		if pullRequest == nil {
```

Comment:
I think we should do both: create a PR to delete PaC configuration (if any) and close existing PaC PR (if any).